Support array of U8 in seeds

mcintyre94 opened the pull request:
Anchor supports a (reference to an) array of `U8` values as a seed value. Using this allows the client-side code to be more straightforward than having each value as its own seed, since we can pack the array into a single buffer: `Buffer.from([x, y])`.

This PR adds support for this to Seahorse, such that this becomes valid:

```python
@instruction
def create_pixel(
  pixel: Empty[Pixel],
  user: Signer,
  pos_x: u8,
  pos_y: u8
):
  # Initialise pixel account
  pixel_account = pixel.init(
    payer = user,
    seeds = ["pixel", [pos_x, pos_y]]
  )
```

The generated accounts context has seeds:

```rust
#[derive(Accounts)]
# [instruction (pos_x : u8 , pos_y : u8)]
pub struct CreatePixel<'info> {
    #[account(
        init,
        payer = user,
        seeds = ["pixel".as_bytes().as_ref(), [pos_x, pos_y].as_ref()],
        bump,
        space = 8 + std::mem::size_of::<Pixel>()
    )]
    pub pixel: Box<Account<'info, Pixel>>,
    #[account(mut)]
    pub user: Signer<'info>,
    pub system_program: Program<'info, System>,
}
```

The PDA can then be generated in JS:

```js
const [pixelPublicKey] = web3.PublicKey.findProgramAddressSync(
  [Buffer.from("pixel"), Buffer.from([x, y])],
  program.programId,
)
```

From experimenting with Anchor I think only an array of U8 needs to be supported. A string/pubkey/longer number would take multiple bytes and then you'd have a nested list which Anchor doesn't seem to like. So I think it's only Array of U8 that should be supported. 

Closes #4

ameliatastic commented:
hey this looks great!

small nit - can you change it to just use the SeedsUnsupportedType for the error instead of introducing a new error type specifically for arrays? after all, arrays are just another type haha.

mcintyre94 commented:
> hey this looks great!
> 
> small nit - can you change it to just use the SeedsUnsupportedType for the error instead of introducing a new error type specifically for arrays? after all, arrays are just another type haha.

Yep sure, updated it! I refactored a little bit to make the outer `ty` available in that error message, so you get eg. `Hint: initializer/signer seeds may be other accounts, strings, integers and arrays of U8. Found: Array(String, Any)` if you try to use an array of the wrong type.

ameliatastic commented:
looks good! this will be included in the next patch :) thanks for contributing!
